src/RedisDiscover.cpp

```diff
@@ -0,0 +1,83 @@
+#include "RedisDiscover.h"
+ 
+RedisDiscover::~RedisDiscover() {
+    redisFree(redis_client_);
+}
+
+std::string RedisDiscover::getRedisHostNameFromEnv() {
+    // TODO: read URL/Hostname from env variable
+    return "redis";
+}
+
+void RedisDiscover::connect() {
+    struct timeval timeout = { 1, 500000 };
+    int port = 6379;  // default Redis port
+    std::string hostname = getRedisHostNameFromEnv();
+
+    redis_client_ = redisConnectWithTimeout(hostname.c_str(), port, timeout);
+    if (redis_client_ == NULL || redis_client_->err) {
+        if (redis_client_) {
+            char err [128];
+            strcpy(err, redis_client_->errstr);
+            redisFree(redis_client_);
+            throw err;
+        } else {
+            throw "Connection error: can't allocate redis context";
+        }
+    }
+}
+
+// get all tasks available in this network from Redis
+std::vector<std::string> RedisDiscover::getTasks() {
+    std::vector<std::string> tasks;
+
+    redisReply* reply = (redisReply*) redisCommand(redis_client_, "LRANGE %s 0 -1", tasks_list_.c_str());
+
+    if (reply->type == REDIS_REPLY_ARRAY) {
+        for (unsigned int j = 0; j < reply->elements; j++) {
+            tasks.push_back(reply->element[j]->str);
+        }
+    }
+    freeReplyObject(reply);
```

Comment:
you should make c++ wrappers for any c library. the constructor sould call redisCommand, and the destructor would call freeReplyObject. there should be no cleanup function calls in any business logic, only in resource handling libraries.